src/core/unit.c

```diff
@@ -2600,8 +2600,12 @@ void unit_notify(Unit *u, UnitActiveState os, UnitActiveState ns, UnitNotifyFlag
                 unit_check_binds_to(u);
 
                 if (os != UNIT_FAILED && ns == UNIT_FAILED) {
-                        reason = strjoina("unit ", u->id, " failed");
-                        emergency_action(m, u->failure_action, 0, u->reboot_arg, unit_failure_action_exit_status(u), reason);
+                        if (m->objective != MANAGER_SHUTDOWN) {
+                                reason = strjoina("unit ", u->id, " failed");
+                                emergency_action(m, u->failure_action, 0, u->reboot_arg, unit_failure_action_exit_status(u), reason);
+                        } else {
+                                log_info( "Tried to add Failure Action reboot job - Shutdown active" );
+                        }
```

Comment:
Most of the other modifications in `src/core` are unexpected, and I think a localized change here is all that should be needed, or actually might be even nicer to modify emergency_action().

Basically, you want to have a condition like
```C
if (IN_SET(m->objective, MANAGER_REBOOT, MANAGER_POWEROFF, MANAGER_HALT, MANAGER_KEXEC))
    /* log a line that the emergency action is skipped because the manager is already terminating... */
else {
    /* do the old thing */
}
```